src/app/cli/src/coda.ml

```diff
@@ -520,16 +559,19 @@ let daemon logger =
          in
          trace_database_initialization "consensus local state" __LOC__
            trust_dir ;
+         let net_ivar = Ivar.create () in
          let net_config =
            { Coda_networking.Config.logger
            ; trust_system
            ; time_controller
            ; consensus_local_state
+           ; chain_id= Lazy.force chain_id
            ; gossip_net_params=
                { timeout= Time.Span.of_sec 3.
                ; logger
                ; target_peer_count= 8
                ; conf_dir
+               ; filter_peer= make_filter_peer (Ivar.read net_ivar) logger
```

Comment:
That's where I put it at first, although I ran into an issue that I figured out how to fix this time around.  Due to where gossip_net sits in the dep graph it needs the chain ID to be supplied externally (can't reference `coda_state` because `consensus`, one of its deps, depends on `gossip_net`). I figured if the abstraction was going to leak anyway I may as well make it a proper callback.

But I think the change is good. Done.